python/cuml/common/base.pyx

```diff
@@ -364,6 +368,24 @@ class Base(TagsMixin,
             return {'preserves_dtype': [self.dtype]}
         return {}
 
+    def _set_nvtx_annotations(self):
+        nvtx_benchmark = os.getenv('NVTX_BENCHMARK')
+        if nvtx_benchmark and nvtx_benchmark.lower() == 'true':
+            self.set_nvtx_annotations()
+
+    def set_nvtx_annotations(self):
+        for func_name in ['fit', 'transform', 'predict', 'fit_predict',
+                          'fit_transform']:
+            if hasattr(self, func_name):
+                message = self.__class__.__module__ + '.' + func_name
+                msg = '{class_name}.{func_name} [{addr}]'
+                msg = msg.format(class_name=self.__class__.__module__,
+                                 func_name=func_name,
+                                 addr=hex(id(self)))
+                func = getattr(self, func_name)
+                func = nvtx.annotate(message=msg)(func)
```

Comment:
I think it'd be a good idea to use a domain for cuML https://nvtx.readthedocs.io/en/latest/basic.html#domains similar to how cuDF does https://github.com/rapidsai/cudf/blob/791beb2989906bd48ea743a294ab56e4fb0bf6de/python/cudf/cudf/core/groupby/groupby.py#L119

Not sure about categories, maybe they could help but I don't see immediately how

Reply:
Done